### ISSUE TITLE
Per application : cannot remove properly and overriden configuration

### ISSUE THREAD

**Issue by MTindev** [negesti/gnome-shell-extensions-negesti#184]
## Expected Behavior

I expect, when saving a setting for an app, than it does not override other app settings which are already saved.
I expect to be able to remove an application (and its settings) without the extension removes the previous or next selected application or refuse to remove one.
I expect a way to reset all configuration. I expected the whole configuration to be deleted when removing (delete) the gnome shell extension from my computer, before installing it again and getting a new fresh install, but setttings are kept.

N.B. I'm on a new Ubuntu 20.04 fresh install, with this issue. That worked like a charm on Ubuntu 19.10.

## Current Behavior

Result : the first application inherits of the second setting.
I added several applications, saved one by one, and one in two has this issue, a kind of override of the previous/next one.
In my case, that happened with "Thunderbird" and "Firefox", "Chrome" and "Org" (Nautilus).

When removing an application, the message tells me "Are you sure to delete the configuration of <APP> ?", where application name is not the selected one but the last one which has already been deleted.
When only one app is kept, the "Org" (Nautilus) one, the deletion popin detects it as "NULL", with the message : "Are you sure to delete the configuration of null ?" and the extension can not remove it from a list.

I removed this gnome shell extension (disable then delete), and install it again : the previous configuration is kept, so I can not try another set of setting. "Org" (Nautilus) is still in the list.

## Possible Solution

I'm suspecting a melting of saved rows in configuration, that could explain than one app setting is set to another, than the deletion do not use the expected row, than the last one app setting can not be removed.

## Steps to Reproduce (for bugs)

1. I added one application, set it settings and saved it.
2. I added a second  application, set it settings and saved it.
3. I added a third application, set it settings and saved it, but that override second application settings.
4. I added a fourth application, set it settings and saved it.
3. I added a fifth application, set it settings and saved it, but that override the fourth application settings.

## Your Environment

* gnome-shell version: 3.36.3
* affected application(s) (e.g. chrome, nautilus, all, ...) : in my case, settings saving for Thunderbird, Firefox, Chrome, Nautilus (and probably other ones).
* distribution : Ubuntu 20.04, a new fresh install.

Exemple for the last row to delete :
![gnome_extension_put_windows_melded_config](https://user-images.githubusercontent.com/988593/97155390-94cce300-1775-11eb-9f04-a57346fe613c.png)

**Comment by MTindev** [comment]
New attempt, same issue for the overridden behavior.

To remove gnome extension settings, I remove the extension and via dconf-editor I removed the keys linked to the extension, then did a reboot.

I reinstalled the gnome extension and had the same issue during configuration, with the following steps :
1. Open configuration panel for the extension, add "Firefox" and set its settings, then save, then close configuration windows.
2. Reopen configuration panel for the extension, add "Mattermost" and set its settings, then save, then clos configuration windows.
3. Reopen configuration panel for the extension, add "Thunderbird" and set its settings, then save, then close configuration windows.

Result : Second app ("Mattermost") settings are overriden by the third one ("Thunderbird").

For removing them, this time no issue : I was able to remove them, from the last one to the first : no row is kept with a "null" error.
